fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt

```diff
@@ -96,15 +102,51 @@ object MatcherHelper {
     ) {
         for (appItem in appItems) {
             if (exists) {
+                Log.i(TAG, "assertUIObjectExists: Trying to verify ${appItem.selector} exists")
                 assertTrue("${appItem.selector} does not exist", appItem.waitForExists(waitingTime))
                 Log.i(TAG, "assertUIObjectExists: Verified ${appItem.selector} exists")
             } else {
+                Log.i(TAG, "assertUIObjectExists: Trying to verify ${appItem.selector} does not exist")
                 assertFalse("${appItem.selector} exists", appItem.waitForExists(waitingTimeShort))
                 Log.i(TAG, "assertUIObjectExists: Verified ${appItem.selector} does not exist")
             }
         }
     }
 
+    fun assertComposeItemExists(
+        vararg composeItems: SemanticsNodeInteraction,
+        exists: Boolean = true,
+    ) {
+        for (composeItem in composeItems) {
+            if (exists) {
+                Log.i(TAG, "assertComposeItemExists: Trying to verify compose item ${composeItem.printToString()} exists")
+                composeItem.assertExists()
+                Log.i(TAG, "assertComposeItemExists: Compose item ${composeItem.printToString()} exists")
+            } else {
+                Log.i(TAG, "assertComposeItemExists: Trying to verify compose item ${composeItem.printToString()} does not exist")
+                composeItem.assertDoesNotExist()
+                Log.i(TAG, "assertComposeItemExists: Compose item ${composeItem.printToString()} does not exist")
+            }
+        }
+    }
+
+    fun assertEspressoItemIsDisplayed(
+        vararg espressoItems: ViewInteraction,
+        exists: Boolean = true,
+    ) {
+        for (espressoItem in espressoItems) {
+            if (exists) {
+                Log.i(TAG, "assertEspressoItemExists: Trying to verify espresso item $espressoItem is displayed")
+                espressoItem.check(matches(isDisplayed()))
```

Comment:
There are many situations where we use a UIObject wait method before doing an espresso assertion or action. One example below: 
<img width="749" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/4037709/13a71c91-fa4f-418e-8bb5-d74f7030cb7f">

@jjSDET How should we handle these situations, where we need to make sure we wait long enough for the item to be visible? Should we just replace everything Espresso with UIAutomator? (I don't like that idea, as it turned out to work better sometimes than UI Automator)